### PR TITLE
Add a "Best Practices" section to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by dennishuo**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,3 +33,39 @@ accept your pull requests.
...
+## Best Practices
+
+The following best-practice guidelines will help ensure your initialization
+actions are less likely to break from one Dataproc version to another, and
+most likely to support different single-node, high-availability, and
+standard cluster modes.
+
+1. Where possible, install from Dataproc's prebuilt Debian packages (built
+   using Apache Bigtop) instead of installing from tarballs. Dataproc packages
+   can be found under /var/lib/apt/lists/*dataproc-bigtop-repo_*Packages on
```
Comment (jmikula) [inline CONTRIBUTING.md]: Should they install them from here or should they rely on `apt-get install <package>`?

Comment (jmikula) [inline CONTRIBUTING.md]: the \*dataproc-bigtop-repo_\* is not rendering in italics. Also, do you want a space before Packages?

Comment (dennishuo) [reply CONTRIBUTING.md]: Added clarification. Indeed they should just use "apt-get install", I just wanted to explain that apt-get install grabs from Dataproc's distro (resolving conflicts between some other Hadoop-related debian package vs Dataproc's sources could be a more advanced matter to explain).

Comment (dennishuo) [reply CONTRIBUTING.md]: "Packages" is part of the glob. Added to whole thing to backticks now.

**Pull request merged**